ddtrace/appsec/_asm_request_context.py

```diff
@@ -122,25 +113,25 @@ def __init__(self):
 
         self._id = _DataHandler.main_id
         self.active = True
-        self.token = _ASM.set(env)
+        self.execution_context = core.ExecutionContext(__name__, **{"asm_env": env})
```

Comment:
That's right.

Reply:
In that case, what's the reason for calling `dispatch` in `ExceutionContext` `__init__`? I get that it's nice to have the debug log about creating a new context, but is there another reason? 

Is it functionality that we want to create that's kind of like `on_span_start`, where we now have the option to add a listener that can run on a certain ExecutionContext init?

Reply:
Yup, you've got it. It's not listened to yet, but [in the future](https://github.com/DataDog/dd-trace-py/pull/6331/files#diff-fdcc9c56641c65611b0d8103b5f98fceaa094149d630d8c755877238cba11485R372) it will be.